fili-core/src/main/java/com/yahoo/bard/webservice/data/filterbuilders/DruidInFilterBuilder.java

```diff
@@ -0,0 +1,80 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.data.filterbuilders;
+
+import com.yahoo.bard.webservice.data.dimension.Dimension;
+import com.yahoo.bard.webservice.data.dimension.DimensionRowNotFoundException;
+import com.yahoo.bard.webservice.druid.model.filter.AndFilter;
+import com.yahoo.bard.webservice.druid.model.filter.Filter;
+import com.yahoo.bard.webservice.druid.model.filter.InFilter;
+import com.yahoo.bard.webservice.druid.model.filter.NotFilter;
+import com.yahoo.bard.webservice.druid.model.filter.OrFilter;
+import com.yahoo.bard.webservice.druid.model.filter.SelectorFilter;
+import com.yahoo.bard.webservice.web.ApiFilter;
+import com.yahoo.bard.webservice.web.FilterOperation;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * A DruidInFilterBuilder builds a conjunction of in-filters for each Dimension, where each in-filter corresponds to a
+ * filter term. So, the filter terms on dimension {@code category}:
+ * <p>
+ * {@code category|id-in[finance,sports],category|desc-contains[baseball]}
+ * <p>
+ * are translated into:
+ * <pre>
+ *     {@code
+ *     {
+ *         "type": "in",
+ *         "dimension": "category",
+ *         "values": ["finance", "sports"]
+ *     }
+ *     }
+ * </pre>
+ * Each filter term is resolved independently of the other filter terms.
+ */
+public class DruidInFilterBuilder extends ConjunctionDruidFilterBuilder {
+    private static final Logger LOG = LoggerFactory.getLogger(DruidInFilterBuilder.class);
+
+    @Override
+    protected Filter buildDimensionFilter(Dimension dimension, Set<ApiFilter> filters)
+            throws DimensionRowNotFoundException {
+        LOG.trace("Building dimension filter using dimension: {} \n\n and set of filter: {}", dimension, filters);
+
+        List<Filter> inFilters = new ArrayList<>();
+        for (ApiFilter filter : filters) {
+            ApiFilter normalizedFilter = filter;
+            if (normalizedFilter.getOperation().equals(FilterOperation.notin)) {
+                normalizedFilter = filter.withOperation(FilterOperation.in);
+            }
+            Filter disjunction = new OrFilter(buildSelectorFilters(
```

Comment:
Why go through all these loops of building a disjunction of selector filters
and then building an inFilter from that?

Why not just get the DimensionRows, and exact the ids from each row, and build
the inFilter directly? Something like:

```java
    List<String> ids = getFilteredDimensionRows(
            dimension,
            Collections.singleton(normalizedFilter)
    )
            .stream()
            .map(DimensionRow::getId)
            .collect(Collectors.toList());
    InFilter filter = new InFilter(dimension, ids);
```